### PR TITLE
feat: Allow gzip compression for static assets.

### PR DESCRIPTION
**Pull request by mturoci**

### CODE REVIEW

**Comment by mturoci** [comment]
Thanks for review @geomodular and pointing me to a more idiomatic Go pattern!

File: gzip.go [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+	New: func() interface{} {
+		return gzip.NewWriter(ioutil.Discard)
+	},
+}
+
+func (w *gzipResponseWriter) Write(b []byte) (int, error) {
+	return w.Writer.Write(b)
+}
+
+func GzipHandler(h http.Handler) http.Handler {
```
Comment (geomodular) [inline gzip.go:40]: nit: I'd make a verb from the function name and start it with a small letter.

**Review by geomodular — APPROVED** [review]
Thanks, @mturoci, looks nice.

File: gzip.go (hunk 1 above)
Comment (mturoci) [reply gzip.go:40]: But starting it with a small letter would prevent exporting it, wouldn't it?

Comment (geomodular) [reply gzip.go]: It depends on the package. The package `wave` is the same for `gzip.go` as for `web_server.go` with no intention to use it elsewhere.

**Review by lo5 — APPROVED** [review]
Thanks @mturoci! Welcome to Go ;-)

**Pull request merged**